### PR TITLE
docs: add docstrings for on_d, on_q, on_tth, and dump

### LINKED ISSUE #306: docs: docstring for `on_d`, `on_q`, `on_tth`, and `dump` are missing
@sbillinge Do we want to make `on_d`, `on_q`, `on_tth` private?

### PR DESCRIPTION
**Pull request by yucongalicechen**
closes #306 
@sbillinge ready for review

### CODE REVIEW

File: src/diffpy/utils/diffraction_objects.py [hunk 1]
```diff
@@ -418,12 +418,33 @@ def _get_original_array(self):
             return self.on_d(), "d"
 
     def on_q(self):
+        """Return the tuple of two 1D numpy arrays containing q and y data.
+
+        Returns
+        -------
+        (xarray, yarray) : tuple of ndarray
```
Comment (sbillinge) [inline src/diffpy/utils/diffraction_objects.py:425]: let's change `xarray` to `q-array` to be explicit.  And also below, xarray-> `tth-array` and so on.  Just to make it really clear.

File: src/diffpy/utils/diffraction_objects.py [hunk 2]
```diff
@@ -507,6 +528,16 @@ def on_xtype(self, xtype):
             raise ValueError(_xtype_wmsg(xtype))
 
     def dump(self, filepath, xtype=None):
+        """Dump the xarray and yarray of the diffraction object to a two-column
+        file, with the associated information included in the header.
+
+        Parameters
+        ----------
+        filepath : str
```
Comment (sbillinge) [inline src/diffpy/utils/diffraction_objects.py:536]: non programmers may not know that that the path includes the filename.  Best here is to give an example.

**Review by sbillinge — COMMENTED** [review]
brilliatn @yucongalicechen .  Please see a couple of small edits.

File: src/diffpy/utils/diffraction_objects.py (hunk 2 above)
Comment (yucongalicechen) [reply src/diffpy/utils/diffraction_objects.py:536]: @sbillinge added in the new commit. please review

File: src/diffpy/utils/diffraction_objects.py [hunk 3]
```diff
@@ -537,6 +537,14 @@ def dump(self, filepath, xtype=None):
             The filepath where the diffraction object will be dumped
         xtype : str, optional, default is q
             The type of quantity for the independent variable chosen from {*XQUANTITIES, }
+
+        Examples
+        --------
+        To save a diffraction object to a file named "diffraction_data.chi" with
+        the independent variable 'q':
+
+        >>> file = "diffraction_data.chi"
```
Comment (sbillinge) [inline src/diffpy/utils/diffraction_objects.py:546]: Let's also have another example, that shows a relative path.  I think that should then be clear enough.

**Review by sbillinge — COMMENTED** [review]
nice!  One last request.

File: src/diffpy/utils/diffraction_objects.py (hunk 3 above)
Comment (yucongalicechen) [reply src/diffpy/utils/diffraction_objects.py:546]: Sounds great! I'll add that.

**Pull request merged**